src/core/qgsvectorlayer.h

```diff
@@ -953,10 +953,18 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer
     virtual QString subsetString();
 
     /**
-     * Query the provider for features specified in request.
+     * Query the layer for features specified in request.
      */
     QgsFeatureIterator getFeatures( const QgsFeatureRequest& request = QgsFeatureRequest() );
 
+    /**
+     * Query the layer for features that match the given filter expressions
+     * @param filterExpression An expression that features on this layer must match to be returned
+     *                         by this iterator
+     * @return A feature iterator
+     */
+    QgsFeatureIterator getFeatures( const QString& filterExpression );
+
```

Comment:
Could we not have an extra getFeatures() method? :) I think it would be confusing that some places support expression directly, while others require it to be wrapped in QgsFeatureRequest... maybe if we had an interface e.g. QgsAbstractFeatureStore (with one virtual getFeatures()) that various classes would implement, such interface could have convenience wrappers like getFeatures( expr ), getFeatures( rect ), getFeature( id ) that would be common to all implementations...